tcms/testruns/tests/test_admin.py

```diff
@@ -22,3 +28,21 @@ def test_regular_user_can_not_change_testrun(self):
         response = self.client.get(reverse('admin:testruns_testrun_change',
                                            args=[self.test_run.pk]))
         self.assertRedirects(response, reverse('testruns-get', args=[self.test_run.pk]))
+
+    def test_regular_user_can_not_delete_testrun(self):
+        response = self.client.get(reverse('admin:testruns_testrun_delete',
+                                           args=[self.test_run.pk]), follow=True)
+        self.assertContains(response, _('Permission denied: TestRun does not belong to you'))
+
+    def test_superuser_can_delete_testrun(self):
+        self.client.login(
+            username=self.superuser.username,
+            password='password')
+        response = self.client.get(reverse('admin:testruns_testrun_delete',
+                                           args=[self.test_run.pk]))
+        self.assertContains(response, _("Yes, I'm sure"))
+        response = self.client.post(reverse('admin:testruns_testrun_delete',
+                                            args=[self.test_run.pk]), {'post': 'yes'}, follow=True)
+        self.assertRedirects(response, reverse('admin:testruns_testrun_changelist'))
+        self.assertNotContains(response, ('<a href="/admin/testruns/testrun/%d/change/">%s</a>'),
```

Comment:
This is missing the parameters for string formatting and will PASS even if the delete fails.

A more accurate way to verify that a record is missing is to query the database directly. Use TestRun.objects.filter() and .exists() methods together with the TR ID which you already have to verify that the object we deleted is not found in the database.